testing/run_tests.sh

```diff
@@ -20,6 +20,9 @@ set -x  # print commands as they are executed
 set -e  # fail and exit on any command erroring
 set -u  # fail and exit on any undefined variable reference
 
+# Get the absolute path to the directory containing this script.
+DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
```

Comment:
Can we just require running the script from the root directory?

Reply:
well, this is just for travis so it probably doesn't matter much, but i'd rather be explicit and safe than make the script more fragiel w.r.t. location.